### PR TITLE
Add missing export for ts declaration

### LINKED ISSUE #508: Typescript declarations are missing in v5
In ember-inflector v4 there were added manuelly all types so that everythng works fine inside a typescript project.

It looks like the types are not anymore present in v5

In v4 there were the types manually added as `index.d.ts` (see https://github.com/emberjs/ember-inflector/blob/v4.0.3-ember-inflector/index.d.ts)

### PR DESCRIPTION
**Pull request by mkszepp**
In v5 there are missing the typescript declarations.

fix #508

### CODE REVIEW

File: ember-inflector/package.json [hunk 1]
```diff
@@ -9,8 +9,14 @@
   "license": "MIT",
   "author": "",
   "exports": {
-    ".": "./dist/index.js",
-    "./*": "./dist/*.js",
+    ".": {
+      "types": "./declarations/index.d.ts",
```
Comment (NullVoxPopuli) [inline ember-inflector/package.json:13]: Why not point these at the original src d.ts?

Comment (mkszepp) [reply ember-inflector/package.json:13]: in this case we need to add in package.json under files also `src` or? and so we do publish to much files or?

Comment (NullVoxPopuli) [reply ember-inflector/package.json:13]: adding `src` under `files` would be sufficient (and common for manual .d.ts)

Comment (mkszepp) [reply ember-inflector/package.json]: changes... is it so correct?

**Review by NullVoxPopuli — APPROVED** [review]

File: ember-inflector/package.json (hunk 1 above)
Comment (NullVoxPopuli) [reply ember-inflector/package.json]: ye

**Pull request merged**